fapolicy_analyzer/ui/profiler_page.py

```diff
@@ -90,7 +119,7 @@ def get_entry_dict(self):
             "dirText": self.get_object("dirEntry").get_text(),
             "envText": self.get_object("envEntry").get_text(),
         }
-
+        dispatch(set_profiler_state(entryDict))
```

Comment:
If you implement the class level `profilerDict ` like suggested above you can set its value here and then dispatch it. That way in your `on_next_system` code the class object and the one coming in on system should be equal and you it will eliminate the extra refresh of the form data.